src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp

```diff
@@ -94,6 +102,53 @@ double get_temporal_id_value(const T& id) {
   }
 }
 
+template <typename InterpolationTargetTag, typename TemporalId>
+std::string interpolator_output_prefix(const TemporalId& temporal_id) {
```

Comment:
These should return the `std::stringstream` instead so whatever else can take advantage of the scientific precision. Or just take a `std::stringstream&` as an argument.